File: src/Core/Models/SystemInfo.cs
namespace CyberPatriotAutomation.Core.Models;

/// <summary>
/// Represents current system state information
/// </summary>
public class SystemInfo
{
    public string? OSVersion { get; set; }
    public List<string> RunningServices { get; set; } = new();
    public List<string> InstalledApplications { get; set; } = new();
    public List<string> UserAccounts { get; set; } = new();
    public List<string> FirewallRules { get; set; } = new();
    public Dictionary<string, string> RegistrySettings { get; set; } = new();

    // Added for audit task output
    public string? RawOutput { get; set; }
    public string? ErrorOutput { get; set; }
}

File: rust/src/models/system_info.rs
use std::collections::HashMap;

/// Represents current system state information.
#[derive(Debug, Default, Clone)]
pub struct SystemInfo {
    pub os_version: Option<String>,
    pub running_services: Vec<String>,
    pub installed_applications: Vec<String>,
    pub user_accounts: Vec<String>,
    pub firewall_rules: Vec<String>,
    pub registry_settings: HashMap<String, String>,

    // Added for audit task output
    pub raw_output: Option<String>,
    pub error_output: Option<String>,
}

impl SystemInfo {
    pub fn new() -> Self {
        Self::default()
    }
}
